woocommerce/woocommerce
# Add beta tester tool to force the coming soon landing pages to display on the front-end

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/48655

Add a tool to WooCommerce Beta Tester to force the coming soon landing pages to show regardless of the site visibility settings.

![Screenshot 2024-07-10 at 19 52 01](https://github.com/woocommerce/woocommerce/assets/4344253/0038ca93-f7ce-47ff-9d4d-9b126c392fcd)




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch locally.
2. cd to `plugins/woocommerce-beta-tester` and run `pnpm run build:zip`
or using [woocommerce-beta-tester.zip](https://github.com/user-attachments/files/16188838/woocommerce-beta-tester.zip)
3. Create a new JN site with the latest WooCommerce 
4. Upload and activate `woocommerce-beta-tester.zip`
5. Go to `/wp-admin/admin.php?page=wc-settings&tab=site-visibility`
6. Update the site visibility to `Live`
7. Go to `Tools -> WCA Test Helper -> Tools`
8. Select `Whole Site` or `Store Only` and click on `Run
9. Confirm it ran successfully
10. Open an incognito window and visit the site front end
11. Confirm the coming soon landing page is shown based on the mode selected in step 8
12. Go to Site Visibility Settings and confirm the site visibility is still set to `Live`
13. Go to task list and confirm the LYS task is marked as complete
14. Play around with the site and confirm everything works as expected



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
